### ISSUE TITLE
Bug: Boost Union blocks Moodle 4.0.4+ (20221108)

### ISSUE THREAD

**Issue by ralf-krause** [moodle-an-hochschulen/moodle-theme_boost_union#133]
I got an error when I updated my MoodleBox up to Moodle 4.0.4+ (20221108).

```
Coding error detected, it must be fixed by a programmer: 
Cannot set custom filetypes because they are defined in config.php
```

First I thought that the error is specific for the MoodleBox https://moodlebox.net . I updated two other platforms to the same Moodle version. On my MoodleBox it was not possible to open the web administration anymore.

After I changed the parameter customfiletypes to xcustomfiletypes I could open the web administration again. Yes, I know that it does not make sense to change a parameter name. I did this to find a solution for the problem. In the MoodleBox config.php there is a part which defines a custom file type.

```
$CFG->customfiletypes = array(
  (object)array(
    'extension' => 'crt',
    'icon' => 'sourcecode',
    'type' => 'application/x-x509-ca-cert',
    'customdescription' => 'X.509 CA certificate'
  )
);
```

Nicolas Martignoni who is the developer of the MoodleBox did exactly what he should do to define a custom file type. Please look into this [comment in config-dist.php](https://github.com/moodle/moodle/blob/5877ae306efaa93a3a0cf8adcedee4da405677f5/config-dist.php#L613-L628).
You also can read https://github.com/moodlebox/moodlebox/issues/273

Okay, I should create a tracker issue for this because the problem is the result of the Moodle update from 4.0.4+ (20221104) to 4.0.4+ (20221108).
https://tracker.moodle.org/browse/MDL-76247

No, the problem doesn't come from the Moodle code. Jake Dallimore used that exact configuration and had been able to upgrade just fine to 4.0.4 weekly. He was closing this one as cannot reproduce.

So I took my own distribution Moodle4mac 4.0.4 from https://download.moodle.org/macosx/ ... I just updated to Moodle 4.0.4+ (20221104) last weekend. So I had a good start for testing. I added the customfiletypes definition from the MoodleBox to the config.php. Everything run fine. I updated to Moodle 4.0.4+ (20221108) now. And Moodle runs without any problem.

Now I installed the Boost Union v4.0-r6 theme_boost_union_moodle40_2022080910
Yes, just in this moment the problem came up. No other additional plugin is installed.

I hope you will get the solution as soon as possible. I like to use Boost Union in my MoodleBox and in all my other platforms. The other platforms are running because the don't have the customfiletypes definition in the config.php

**Comment by abias** [comment]
Hi @ralf-krause ,

many thanks for raising this issue and for the detailled bug report. I am really sorry that it affected your Moodle update. This shouldn't have happened.

I could confirm that this issue was triggered by the presence of `$CFG->customfiletypes` in config.php together with the latest version of Boost Union:

* The bug happens as soon as you add a `$CFG->customfiletypes` entry to config.php.
* The bug happened (for me) on Moodle 4.0.4+ (Build: 20221104) already, thus it's not related to your Moodle update.
* The bug does not necessarily happen during a DB update, it just breaks the site administration page (which is shown during the DB update) and maybe some other areas of Moodle badly.
* The bug was introduced with the change in #121 which was released within v4.0-r6 of this plugin.

Thus, you were hit be the theme update, not be the Moodle core update.

#121 introduced some upload restrictions for webfonts filetypes and as Moodle does not know all existing webfonts filetypes, we registered them on-the-fly on the settings page. However, we learnt know that Moodle does not allow us to register new filetypes if they are already set in config.php. That does not fully make sense, but it's a fact which Boost Union will have to respect.

I implemented a fix for this issue (which simply does not register the filetypes as soon as `$CFG->customfiletypes` is set in config.php) and integrated it in #134.

A new release on moodle.org will follow soon.

Thanks again for the report and sorry for the inconvenience.

**Comment by ralf-krause** [comment]
Great! Thanks!
